sycl/include/sycl/detail/vector_arith.hpp

```diff
@@ -265,7 +265,8 @@ template <typename Self> struct VecOperators {
           // ensure we generate 0/1 only (and not 2/-1/etc.).
 #if __clang_major__ >= 20
           // Not an integral constant expression prior to clang-20.
-          static_assert((ext_vector<int8_t, 2>{1, 0} == 0)[1] == -1);
+          static_assert(
+              static_cast<int8_t>((ext_vector<int8_t, 2>{1, 0} == 0)[1]) == -1);
```

Comment:
I think a better fix would be to use the same expression here and in line 272 (whatever it would be). @hvdijk any thoughts on that?

Reply:
Ideally yes, but `ext_vector` is only available on Clang, not with GCC, so cannot be used below, and the below `reinterpret_cast` makes the expression non-constant so it cannot be used in `static_assert`.